File: leetcode2678/solution.cpp
/*
Solution 1:

Time: O(n) | Space: O(1)

- n: length of 'details'
*/

class Solution {
public:
    int countSeniors(vector<string>& details) {
        int res = 0;

        for (string d: details) {
            int s = (d[11] - '0') * 10 + d[12] - '0';
            res += s > 60;
        }

        return res;
    }
};

File: leetcode2678/solution.rs
/*
Solution 1:

Time: O(n) | Space: O(1)

-n: length of details
*/

impl Solution {
    pub fn count_seniors(details: Vec<String>) -> i32 {
        let mut res: i32 = 0;

        for d in details.iter() {
            let d: Vec<u8> = d.bytes().collect::<Vec<u8>>();
            let age: u8 = (d[11] - 48) * 10 + d[12] - 48;

            res += if (age > 60) {
                1
            } else {
                0
            };
        }

        res
    }
}